Source/JavaScriptCore/heap/MarkedBlock.h

```diff
@@ -410,13 +412,16 @@ class MarkedBlock {
     ~MarkedBlock();
     Atom* atoms();
         
-    JS_EXPORT_PRIVATE void aboutToMarkSlow(HeapVersion markingVersion);
+    JS_EXPORT_PRIVATE void aboutToMarkSlow(HeapVersion markingVersion, HeapCell*);
     void clearHasAnyMarked();
     
     void noteMarkedSlow();
     
     inline bool marksConveyLivenessDuringMarking(HeapVersion markingVersion);
     inline bool marksConveyLivenessDuringMarking(HeapVersion myMarkingVersion, HeapVersion markingVersion);
+
+    // This is only used for debugging. We should remove this once the issue is resolved (https://rdar.apple.com/126213131)
+    NEVER_INLINE void dumpInfoIfHandleIsNotValid(AbstractLocker&, HeapCell*);
```

Comment:
Let's call this `dumpInfoAndCrashForInvalidHandle` to make it clear that we're not returning.  You probably should add `NO_RETURN_DUE_TO_CRASH` here too.